src/uct/ib/base/ib_iface.c

```diff
@@ -236,13 +239,30 @@ ucs_status_t uct_ib_iface_recv_mpool_init(uct_ib_iface_t *iface,
                         config->rx.mp.max_bufs);
     }
 
+    status = uct_iface_param_am_alignment(params, UCS_SYS_CACHE_LINE_SIZE, 0,
+                                          &alignment, &align_offset,
+                                          iface->config.seg_size);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    if (params->field_mask & UCT_IFACE_PARAM_FIELD_AM_ALIGNMENT) {
```

Comment:
check alignment==1

Reply:
why? here alignment is user-defined

Reply:
can we check if the alignment passed from API is 1, either as default, or user-defined as 1?

Reply:
i did not get it, can you please elaborate?

Reply:
i mean that we can treat the "user alignment of 1" case same way as "no alignment specified by user"

Reply:
ok, got it

Reply:
btw, do you think it is worth separate handling? It would add more code, but no difference in semantic

Reply:
can we just check 'alignment==1' ?

Reply:
no, if user did not specify alignment we use cache_line_size

Reply:
IMO, if user specified alignment ==1 we can also use cache line size